index.js

```diff
@@ -106,9 +106,19 @@ module.exports._flush = function(lastTime, code) {
 
   return RSVP.Promise.resolve(lastTime).
     then(function() {
-      return RSVP.Promise.all(work.map(function(handler) {
-        return handler.call(null, code);
-      }));
+      var firstRejected;
+      return RSVP.allSettled(work.map(function(handler) {
+        return RSVP.resolve(handler.call(null, code)).catch(function(e) {
+          if (!firstRejected) {
+            firstRejected = e;
+          }
+          throw e;
+        });
+      })).then(function(results) {
+        if (firstRejected) {
+          throw firstRejected;
+        }
+      });
```

Comment:
previously this promise chain returned the fulfilled handler return values, it no longer does. 

I believe that can be solved by:

```js
return result.map(x => x.value); 
```

in this last `then` `fulfillment` handler.